domain/src/main/kotlin/no/nav/su/se/bakover/domain/Søknad.kt

```diff
@@ -14,12 +15,21 @@ data class Søknad(
     sealed class Lukket {
         abstract val tidspunkt: Tidspunkt
         abstract val saksbehandler: Saksbehandler
-        abstract val begrunnelse: String
+        abstract val typeLukking: TypeLukking
 
         data class Trukket(
             override val tidspunkt: Tidspunkt,
             override val saksbehandler: Saksbehandler,
-            override val begrunnelse: String
+            override val typeLukking: TypeLukking = TypeLukking.Trukket
         ) : Lukket()
     }
+
+    data class LukketSøknadBody(
+        val datoSøkerTrakkSøknad: LocalDate,
+        val typeLukking: TypeLukking
+    )
+
+    enum class TypeLukking(val value: String) {
```

Comment:
Det gir ofte ikke mening for domenelaget å ha både en enum og en sealed class for å spesifisere det samme. Jeg foreslår at du mapper denne i brevservice eller typene som tilhører journalting dersom du trenger

Reply:
👍